packages/dds/tree/src/test/simple-tree/toMapTree.spec.ts

```diff
@@ -50,184 +64,271 @@ describe("toMapTree", () => {
+			const actual = nodeDataToMapTree(tree, [schema]);
 
-		const expected: MapTree = {
-			type: brand("test.list"),
-			fields: new Map<FieldKey, MapTree[]>([
-				[
-					EmptyKey,
+			const expected: MapTree = {
+				type: brand("test.list"),
+				fields: new Map<FieldKey, MapTree[]>([
 					[
-						{ type: leaf.number.name, value: 42, fields: new Map() },
-						{ type: leaf.handle.name, value: handle, fields: new Map() },
-						{ type: leaf.number.name, value: 37, fields: new Map() },
+						EmptyKey,
+						[
+							{
+								type: numberSchemaIdentifier,
+								value: 42,
+								fields: new Map(),
+							},
+							{
+								type: handleSchemaIdentifier,
+								value: handle,
+								fields: new Map(),
+							},
+							{
+								type: brand(childObjectSchema.identifier),
+								fields: new Map<FieldKey, MapTree[]>([
+									[
+										brand("name"),
+										[
+											{
+												type: stringSchemaIdentifier,
+												value: "Jack",
+												fields: new Map(),
+											},
+										],
+									],
+									[
+										brand("age"),
+										[
+											{
+												type: numberSchemaIdentifier,
+												value: 37,
+												fields: new Map(),
+											},
+										],
+									],
+								]),
+							},
+						],
 					],
-				],
-			]),
-		};
+				]),
+			};
 
-		assert.deepEqual(actual, expected);
-	});
+			assert.deepEqual(actual, expected);
+		});
 
-	it("list (empty)", () => {
-		const schemaBuilder = new SchemaBuilder({ scope: "test" });
-		const rootSchema = schemaBuilder.list("list", schemaBuilder.number);
-		const schema = schemaBuilder.intoSchema(rootSchema);
+		it("Empty list", () => {
+			const schemaFactory = new SchemaFactory("test");
+			const schema = schemaFactory.array("list", schemaFactory.number);
 
-		const tree: number[] = [];
+			const tree: number[] = [];
 
-		const actual = nodeDataToMapTree(tree, schema, schema.rootFieldSchema.allowedTypeSet);
+			const actual = nodeDataToMapTree(tree, [schema]);
 
-		const expected: MapTree = {
-			type: brand("test.list"),
-			fields: new Map<FieldKey, MapTree[]>(),
-		};
+			const expected: MapTree = {
+				type: brand("test.list"),
+				fields: new Map<FieldKey, MapTree[]>(),
+			};
 
-		assert.deepEqual(actual, expected);
+			assert.deepEqual(actual, expected);
+		});
 	});
 
-	it("map (non-empty)", () => {
-		const schemaBuilder = new SchemaBuilder({ scope: "test" });
-		const rootSchema = schemaBuilder.map("map", [
-			schemaBuilder.number,
-			schemaBuilder.string,
-			schemaBuilder.null,
-		]);
-		const schema = schemaBuilder.intoSchema(rootSchema);
-
-		const entries: [string, InsertableContent][] = [
-			["a", 42],
-			["b", "Hello world"],
-			["c", null],
-			["d", undefined as unknown as InsertableContent], // Should be skipped in output
-		];
-		const tree = new Map<string, InsertableContent>(entries);
+	describe("map", () => {
+		it("Non-empty map", () => {
```

Comment:
again this seems more complex than necessary. I think seperate tests for covering the addation cases would make more sense than lumping it all in "non-empty"

Reply:
https://dev.azure.com/fluidframework/internal/_workitems/edit/7597